### PR TITLE
use JdbcDatabase for snowflake

### PR DESCRIPTION
**Pull request by cgardens**
## What
* I want to massage our destination abstractions. A good prerequisite step is getting the destinations to use JdbcDatabase where we can. This PR just moves snowflake to user the JdbcDatabase abstraction only.

### CODE REVIEW

File: airbyte-db/src/main/java/io/airbyte/db/jdbc/DefaultJdbcDatabase.java [hunk 1]
```diff
@@ -41,15 +41,19 @@
...
-  private final DataSource ds;
+  private final CloseableConnectionSupplier connectionSupplier;
 
-  public DefaultJdbcDatabase(final DataSource ds) {
-    this.ds = ds;
+  public DefaultJdbcDatabase(final DataSource dataSource) {
+    this(new DataSourceConnectionSupplier(dataSource));
+  }
+
+  public DefaultJdbcDatabase(final CloseableConnectionSupplier connectionSupplier) {
```
Comment (cgardens) [inline airbyte-db/src/main/java/io/airbyte/db/jdbc/DefaultJdbcDatabase.java:50]: we don't get a datasource with snowflake so we need to wrap if we want to reuse the DefaultJdbcDatabase.

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java [hunk 2]
```diff
@@ -45,86 +41,8 @@
...
-   * method waits until the result set is ready to be processed.
-   * <p>
-   * This must be called while the connection for the result set is still open.
-   *
-   * @param resultSet Snowflake result set
-   * @return Snowflake result set that is ready for consumption.
-   * @throws InterruptedException - thrown on interruption
-   * @throws SQLException - thrown on unexpected query status
-   */
-  public static ResultSet waitForQuery(ResultSet resultSet) throws InterruptedException, SQLException {
```
Comment (cgardens) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java:62]: i don't think we need any async behavior from snowflake in our impl

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java [hunk 3]
```diff
@@ -150,16 +149,16 @@ public String insertIntoFromSelectQuery(String schemaName, String srcTableName,
     @Override
     public void executeTransaction(String queries) throws Exception {
       String renameQuery = "BEGIN;\n" + queries + "COMMIT;";
-      SnowflakeDatabase.executeSync(connectionFactory, renameQuery, true, rs -> null);
+      database.execute(renameQuery);
```
Comment (cgardens) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java:152]: i need to double check if this is going to work or if i need to alter the snowflake jdbc instance to accept multiple statemtns

**Review by michel-tricot — APPROVED** [review]
Clean

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java (hunk 3 above)
Comment (cgardens) [reply airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java:152]: changed the connection to allow multiple statements.

**Review by michel-tricot — APPROVED** [review]

**Pull request merged**